ocaml/xapi/helpers.ml

```diff
@@ -278,21 +278,15 @@ let call_api_functions_internal ~__context f =
     session
   in
   let session_id =
-    try
-      if Pool_role.is_master() then
-        begin
-          let session_id = Context.get_session_id __context in
-          if Db.Session.get_pool ~__context ~self:session_id
-          then session_id
-          else do_master_login ()
-        end
-      else
-        let session_id = Context.get_session_id __context in
-        (* read any attr to test if session is still valid *)
```

Comment:
Please keep this comment on top of `let in_pool ...`, it clarifies why the function call can fail

Reply:
Yes, we should keep it.